climada/util/coordinates.py

```diff
@@ -459,6 +460,136 @@ def get_gridcellarea(lat, resolution=0.5, unit="ha"):
     return area
 
 
+def compute_grid_cell_area_validation(res: float) -> tuple[np.ndarray]:
+    """
+    This function computes the area of each grid cell on a sphere (Earth), using latitude and
+    longitude bins based on the given resolution. The area is computed using the spherical cap
+    approximation for each grid cell. The function return a 2D matrix with the corresponding area.
+
+    The formula used to compute the area of each grid cell is derived from the integral of the
+    surface area of a spherical cap between squared latitude and longitude bins:
+
+    A = R**2 * (Δλ) * (sin(ϕ2) - sin(ϕ1))
+
+    Where:
+    - R is the radius of the Earth (in km).
+    - Δλ is the width of the grid cell in terms of longitude (in radians).
+    - sin(ϕ2) - sin(ϕ1) is the difference in the sine of the latitudes, which
+    accounts for the varying horizontal distance between longitudinal lines at different latitudes.
+
+    This formula is the direct integration over λ and ϕ2 of:
+
+    A = R**2 * Δλ * Δϕ * cos(ϕ1)
+
+    which approximate the grid cell area as a square computed by the multiplication of two
+    arc legths, with the logitudal arc length ajdusted by latitude.
+
+    Parameters:
+    ----------
+    res: int
+        The resolution of the grid in degrees. The grid will have cells of size `res x res` in
+        terms of latitude and longitude.
+
+    Returns:
+    -------
+    grid_area: np.ndarray
+        A 2D array of shape `(len(lat_bins)-1, len(lon_bins)-1)` containing the area of each grid
+        cell, in square kilometers. Each entry in the array corresponds to the area of the
+        corresponding grid cell on Earth.
+
+    Example:
+    --------
+    >>> res = 10  #10° resolution
+    >>> area = compute_grid_cell_area(res = res)
+    >>> print(area.shape) # (180/10, 360/10)
+    (18, 36)
+    """
+
+    lat_bins = np.linspace(-90, 90, int(180 / res))  # lat bins
+    lon_bins = np.linspace(-180, 180, int(360 / res))
+
+    r_earth = 6371  # Earth's radius [km]
+    # Convert to radians
+    lat_bin_edges = np.radians(lat_bins)
+    lon_res_rad = np.radians(res)
+
+    # Compute area
+    areas = (
+        r_earth**2
+        * lon_res_rad
+        * np.abs(np.sin(lat_bin_edges[1:]) - np.sin(lat_bin_edges[:-1]))
+    )
+    # Broadcast to create a full 2D grid of areas
+    grid_area = np.tile(
+        areas[:, np.newaxis], (1, len(lon_bins) - 1)
+    )  # each row as same area
+
+    return grid_area, lat_bins, lon_bins
+
+
+def compute_grid_cell_area(
+    res: float = None,
+    bounds: tuple = None,
+    projection: str = "WGS84",
```

Comment:
Does this also accept projections for instance in meters? Please make clear if not, and catch the error accordingly.